master/buildbot/config.py

```diff
@@ -541,6 +541,10 @@ def load_slaves(self, filename, config_dict):
             if sl.slavename in ("debug", "change", "status"):
                 msg = "slave name '%s' is reserved" % sl.slavename
                 error(msg)
+            
+            if not util.identifiers.isIdentifier(50, sl.slavename):
+                msg = "Buildslave name %r is not a 50-character identifier" % (sl.slavename,)
+                error(msg)
```

Comment:
Sounds like a good idea: guessing over a cryptic error message is not really a productive way of spending time. :smile: